compiler/AST/symbol.cpp

```diff
@@ -472,12 +472,10 @@ const char* Symbol::getSanitizedMsg(std::string msg) const {
   // TODO: Support explicit title and reference targets like in reST direct hyperlinks (and having only target
   //       show up in sanitized message).
   // TODO: Allow prefixing content with ! (and filtering it out in the sanitized message)
```

Comment:
I think this TODO comment about `!` has been out of date since this PR: https://github.com/chapel-lang/chapel/pull/18796, would you mind removing it?

Reply:
Content that is prefixed with `!` does get filtered but leaves it behind (e.g., ':proc:\`!abc\`' becomes '!abc'), so would it be better to update the TODO rather than remove it?

Reply:
I see (just to repeat what you said): we're now allowing ':proc:\`!abc\`' but not filtering the `!` out when we print the message.  It looks like the purpose of `!` is to prevent a link from being formed (https://chapel-lang.org/docs/latest/tools/chpldoc/chpldoc.html#inline-markup-2) so we should be removing this from our printed messages to the terminal.

#18796 half implemented this by matching `!` but not filtering it out of our terminal messages. Honestly, that seems worse to me than not filtering at all. If the intended message is: 

```
xyz() is deprecated use abc() instead.
```

I'd rather the user be presented with:

```
xyz() is deprecated use :proc:`!abc()` instead.
```

Instead of:

```
xyz() is deprecated use !abc() instead.
```

Obviously just removing the `!` would be the ideal thing to do but if we're going to print a warning message I'd rather it look weird (maybe the user will get tipped off that something is "off") than misleading.

Anyway sorry you're getting roped into this (it looks like I wasn't a reviewer on #18796 so I hadn't put much thought into this until now). I guess there's a few things we could do:

- Ignore it
- Update the comment to say we're currently filtering ':proc:`!abc` to `!abc`, which is wrong and we should also be removing the `!`. It's probably worth mentioning that this is the current status in the Github issue too (https://github.com/chapel-lang/chapel/issues/18549).
- Revert it to not allow matching `!`.
- Implement a fix so we do filter out `!`.

Obviously my preference would be to implement a fix :-) but again you didn't sign up for that so I'd be fine picking it up myself if you'd rather not. If you decide not to I would appreciate updating the comment just so we don't accidentally remove it in the future thinking "oh I see '!' in the regexp, that TODO must have already been taken care of".

Reply:
I might just do the fix (in for a penny in for a pound). I think I have a solution that will solve this cleanly and also handle your other note about malformed strings such as `a!bc` and `a~bc`.